packages/contracts-bedrock/test/dispute/DelayedWETH.t.sol

```diff
@@ -0,0 +1,158 @@
+import { CommonTest } from "test/setup/CommonTest.sol";
+
+contract DelayedWETH_Init is CommonTest {
+    event Approval(address indexed src, address indexed guy, uint256 wad);
+    event Transfer(address indexed src, address indexed dst, uint256 wad);
+    event Deposit(address indexed dst, uint256 wad);
+    event Withdrawal(address indexed src, uint256 wad);
+    event Unwrap(address indexed src, uint256 wad);
+
+    function setUp() public virtual override {
+        super.enableFaultProofs();
+        super.setUp();
+
+        // Transfer ownership of delayed WETH to the test contract.
+        vm.prank(deploy.mustGetAddress("SystemOwnerSafe"));
+        delayedWeth.transferOwnership(address(this));
+    }
+}
+
+contract DelayedWETH_Initialize_Test is DelayedWETH_Init {
+    function test_initialize_succeeds() public {
+        assertEq(delayedWeth.owner(), address(this));
+        assertEq(address(delayedWeth.config()), address(superchainConfig));
+    }
+}
+
+contract DelayedWETH_Unlock_Test is DelayedWETH_Init {
+    function test_unlock_once_succeeds() public {
+        delayedWeth.unlock(alice, 1 ether);
+        (uint256 amount, uint256 timestamp) = delayedWeth.withdrawals(address(this), alice);
+        assertEq(amount, 1 ether);
+        assertEq(timestamp, block.timestamp);
+    }
+
+    function test_unlock_twice_succeeds() public {
+        uint256 ts = block.timestamp;
+        delayedWeth.unlock(alice, 1 ether);
+        (uint256 amount1, uint256 timestamp1) = delayedWeth.withdrawals(address(this), alice);
+        assertEq(amount1, 1 ether);
+        assertEq(timestamp1, ts);
+
+        vm.warp(ts + 1);
+
+        delayedWeth.unlock(alice, 1 ether);
+        (uint256 amount2, uint256 timestamp2) = delayedWeth.withdrawals(address(this), alice);
+        assertEq(amount2, 2 ether);
+        assertEq(timestamp2, ts + 1);
+    }
+
+    function test_unlock_whilePaused_fails() public {
+        address guardian = optimismPortal.GUARDIAN();
+        vm.prank(guardian);
+        superchainConfig.pause("identifier");
+        vm.expectRevert();
+        delayedWeth.unlock(alice, 1 ether);
+    }
+}
+
+contract DelayedWETH_Withdraw_Test is DelayedWETH_Init {
+    function test_withdraw_whileUnlocked_succeeds() public {
+        vm.prank(alice);
+        delayedWeth.deposit{ value: 1 ether }();
+        uint256 balance = address(alice).balance;
+        vm.prank(alice);
+        delayedWeth.unlock(alice, 1 ether);
+        vm.warp(block.timestamp + delayedWeth.delay() + 1);
+        vm.expectEmit(true, true, false, false);
+        emit Withdrawal(address(alice), 1 ether);
+        vm.prank(alice);
+        delayedWeth.withdraw(alice, 1 ether);
+        assertEq(address(alice).balance, balance + 1 ether);
+    }
+
+    function test_withdraw_whileLocked_fails() public {
+        vm.prank(alice);
+        delayedWeth.deposit{ value: 1 ether }();
+        uint256 balance = address(alice).balance;
+        vm.expectRevert();
```

Comment:
I'd suggest adding reasons to the `require` statements in the contracts, then `expectRevert("reason string")` to be stricter—`expectRevert()` with no args passes regardless of revert reason